In MacPorts gtk-update-icon-cache is provided by the gtk2 port.
gtk3 provides gtk-update-icon-cache-3.0 instead

--- setup.py.orig
+++ setup.py
@@ -269,7 +269,7 @@
         if not self.distribution.no_update_icon_cache:
             print("running gtk-update-icon-cache")
             icon_path = os.path.join(self.install_data, "share/icons/hicolor")
-            self.spawn(["gtk-update-icon-cache", "-q", "-t", icon_path])
+            self.spawn(["gtk-update-icon-cache-3.0", "-q", "-t", icon_path])
 
         if not self.distribution.no_compile_schemas:
             print("compiling gsettings schemas")
